core/src/main/java/org/bitcoinj/core/PeerGroup.java

```diff
@@ -1482,9 +1481,9 @@ public synchronized void onBlocksDownloaded(Peer peer, Block block, @Nullable Fi
 
         @Override
         public synchronized void run() {
-            if (blocksInLastSecond > 1) {
```

Comment:
It's >1 deliberately to avoid triggering this path when a long running connection hears about a newly solved block. I guess there should be a comment explaining this.